src/sync.rs

```diff
@@ -50,65 +93,27 @@ impl Lock {
             serde_json::from_str(&contents).unwrap_or_default()
         })
     }
-}
 
-pub fn store_last_state(file_path: &Path, entry: LockEntry) -> Result<(), Box<dyn Error>> {
-    let mut json_lock = Lock::from_file(file_path)?;
+    pub fn store_last_state(&self, file_path: &Path) -> Result<(), Box<dyn Error>> {
```

Comment:
It is also not necessary anymore to pass the `LockEntry` because it is done via the `update`, good